fem/bilinearform_ext.cpp

```diff
@@ -386,6 +387,11 @@ void PABilinearFormExtension::Update()
    bdr_face_restrict_lex = nullptr;
 }
 
+void BilinearFormExtension::UseTensorBasis(const bool use_tensor_basis_)
+{
+   use_tensor_basis = use_tensor_basis_;
+}
```

Comment:
I don't think it would be hard to add some additional checking to avoid mixing tensor and generic integrators. I do want eventually add a sum factorized version.

Two reasons in support of the added functionality are:
1. Meshing well with hex elements on complex geometries isn't always feasible.
2. This is an example with lower barriers to entry for writing the kernels. Elasticity is simple, but for complicated weak forms, it's a big effort for a user to write full and partial assembly, and to also learn the sum factorization stuff, shared memory usage, etc. Maybe there is a generic interface possible to use sum factorization machinery while letting the user work only at the quadrature level.

I think it expands the problems MFEM+GPUs can be used on. But I appreciate the PR needs work since it breaks some assumptions.